dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/ByteArrayPlusOutputStream.java

```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
+ * Copyright (c) 1997, 2004 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.wim.adapter.ldap;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import com.ibm.websphere.ras.annotation.Trivial;
+
+/******************************************************************************
+*	Use an extended version of ByteArrayOutputStream in order to allow access
+*	to protected values.
+******************************************************************************/
+@Trivial
+public class ByteArrayPlusOutputStream extends ByteArrayOutputStream {
+    public final String eol = getSystemProperty("line.separator");
+    @Override
+    public void write(int b) {
+        super.write(b);
+        try {
+            if (this.toString("UTF-8").contains(eol)) {
+                try {
+                    super.writeTo(System.out);
+                    super.reset();
+                } catch (IOException e) {
+                    e.printStackTrace();
```

Comment:
Fix all the e.printStackTrace -- we should trace it, but don't throw an exception. We don't want failing trace to interrupt an ldap reqeust.